### ISSUE TITLE
agent connected with bad signiture holding connection

### ISSUE THREAD

**Issue by mosesBD** [Ylianst/MeshCentral#102]
hi
i tinkered with the UI and changed some files. then i installed a brand new meshcentral server and replaced the files i changed. the changes are just to the user interface not the functionality.
when i try to connect a windows device it gives me this error:
**agent connected with bad signiture holding connection**
i deleted the certs in meshcentral-data directory and ran the server again to rebuild them.
can you please explain how can i replace files that i changed or how to install my own changed version
tnx

**Comment by Ylianst** [comment]
Hi. Sorry for the delay, I had a busy week. UI changes should not cause this. The certificate "agentserver-cert-xxx" in "meshcentral-data" is the certificate that agents are looking for when connecting. If you delete/regenerate that certificate, agent will no longer connect to your server.

If you do reset that certificate, you will have to reinstall the agents. Just download the new agent from your server. When you download the agent, the server injects into the agent a "serverid" that is a SHA384 hash of the "agentserver-cert".

In other words, you can use an agent you downloaded from a server, to connect to a different server... even if it's the same URL.

On the topic of web UI changes... create a folder called "meshcentral-web" (at the same level as "meshcentral-data") then copy the "meshcentral/public" folder into it. Make your changes to files in that folder. MeshCentral will automatically use that folder instead of "public" if it exists and you can now update MeshCentral without your UI changes being erased.

Hope it helps,
Ylian

**Comment by Ylianst** [comment]
I got a private request to add the device group name to the name of the agent when you download it. I think this is a good idea and will close this issue when that is done. Right now, it's a bit confusing what agent will connect a device to what group.

**Comment by mosesBD** [comment]
thank you
if you could please update the meshagent documents so we can compile it and tweak easier, that would be great. for example if i try to translate to a language with utf8 or unicode characters the visual studio sort of breaks the characters.
also is meshconsole a command line version of the client? how do i use it?

**Comment by Ylianst** [comment]
So, it would be best to open a different issue with a more specific requests. For the mesh agent, feel free to post issues on the [MeshAgent issues list](https://github.com/Ylianst/MeshAgent/issues). I just published MeshCentral v0.2.9-e. Now when you download a Windows agent, it will add the name of the device group to the agent. So you have some indication this agent is for this specific group.

MeshConsole is a command line only version, that is really used for debugging and serves as the main agent on Linux and MacOS.

**Comment by Ylianst** [comment]
I will close this bug. Feel free to reopen if a agent gets a bad signature error.
